index.js

```diff
@@ -1,6 +1,61 @@
 /* jshint node: true */
 'use strict';
 
+var path = require('path');
+var Funnel = require('broccoli-funnel');
+var mergeTrees = require('broccoli-merge-trees');
+var VersionChecker = require('ember-cli-version-checker');
+
 module.exports = {
-  name: 'ember-a11y'
+  name: 'ember-a11y',
+
+  init: function() {
+    if (this._super.init) {
+      this._super.init.apply(this, arguments);
+    }
+
+    var versionChecker = new VersionChecker(this);
+    versionChecker.for('ember-cli', 'npm').assertAbove('0.2.0');
+
+    var emberVersion = versionChecker.for('ember', 'bower');
+
+    if ((emberVersion.gt('2.9.0-beta') && emberVersion.lt('2.9.0')) || emberVersion.gt('2.10.0-alpha')) {
+      this.versionSpecificPath = path.join(this.root, 'version-specific-2.9');
+    } else if (!emberVersion.lt('1.13.0')) {
+      this.versionSpecificPath = path.join(this.root, 'version-specific-1.13');
+    } else {
+      throw new Error('This version of ' + this.name + ' supports Ember versions >= 1.13.0.');
+    }
+  },
+
+  treeForAddon: function() {
+    var trees = [this._super.treeForAddon.apply(this, arguments)];
+
+    trees.push(new Funnel(this.versionSpecificPath, {
+      files: ['ember-internals.js'],
+      destDir: 'modules/' + this.name
+    }));
+
+    return mergeTrees(trees, { overwrite: true });
+  },
+
+  treeForTemplates: function() {
+    var trees = [this._super.treeForTemplates.apply(this, arguments)];
+
+    trees.push(this.treeGenerator(
+      path.resolve(this.root, this.versionSpecificPath + '/app/templates')
+    ));
+
+    return mergeTrees(trees.filter(Boolean), { overwrite: true });
+  },
+
+  treeForApp: function(defaultTree) {
```

Comment:
So `_super` can still have side effects so we should always call super unless we're _intentionally_ opting out of it.